operator/standalone/update.go

```diff
@@ -0,0 +1,386 @@
+	return err
+}
+
+func (u *UpdateStandalonePipeline) fetchOperatorConfig(ctx context.Context) error {
+	list := &v1alpha1.PostgresqlStandaloneOperatorConfigList{}
+	labels := client.MatchingLabels{
+		v1alpha1.PostgresqlMajorVersionLabelKey: u.instance.Spec.Parameters.MajorVersion.String(),
+	}
+	ns := client.InNamespace(u.operatorNamespace)
+	err := u.client.List(ctx, list, labels, ns)
+	if err != nil {
+		return err
+	}
+	if len(list.Items) == 0 {
+		return fmt.Errorf("no %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labels, ns)
+	}
+	if len(list.Items) > 1 {
+		return fmt.Errorf("multiple versions of %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labels, ns)
+	}
+	u.config = &list.Items[0]
+	return nil
+}
+
+func (u *UpdateStandalonePipeline) ensureConnectionSecret(ctx context.Context) error {
+	var updateSecretFunc func() error
+	if u.instance.Spec.Parameters.EnableSuperUser {
+		credentialsSecret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{
+			Name:      getCredentialSecretName(),
+			Namespace: u.instance.Status.HelmChart.DeploymentNamespace,
+		}}
+		err := getClientFromContext(ctx).Get(ctx, types.NamespacedName{Name: credentialsSecret.Name, Namespace: credentialsSecret.Namespace}, credentialsSecret)
+		if err != nil {
+			return err
+		}
+		updateSecretFunc = func() error {
+			u.connectionSecret.Data["POSTGRESQL_POSTGRES_PASSWORD"] = credentialsSecret.Data["postgres-password"]
+			return nil
+		}
+	} else {
+		updateSecretFunc = func() error {
+			delete(u.connectionSecret.Data, "POSTGRESQL_POSTGRES_PASSWORD")
+			return nil
+		}
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, u.client, u.connectionSecret, updateSecretFunc)
+	return err
+}
+
+func (u *UpdateStandalonePipeline) ensurePersistentVolumeClaim(ctx context.Context) error {
+	quantity, err := resource.ParseQuantity(u.instance.Spec.Parameters.Resources.StorageCapacity.String())
+	if err != nil {
+		return err
+	}
+	updatePVCFunc := func() error {
+		u.persistentVolumeClaim.Spec.Resources.Requests = map[corev1.ResourceName]resource.Quantity{corev1.ResourceStorage: quantity}
+		return nil
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, u.client, u.persistentVolumeClaim, updatePVCFunc)
+	return err
+}
+
+func (u *UpdateStandalonePipeline) restartPostgresqlPod(ctx context.Context) error {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PostgresqlPodName,
+			Namespace: u.instance.Status.HelmChart.DeploymentNamespace,
+		},
+	}
+
+	// Restart the pod so that the new PVC storage takes effect
+	return u.client.Delete(ctx, pod)
+}
+
+func (u *UpdateStandalonePipeline) isStorageUpdated(_ context.Context) bool {
+	quantity, err := resource.ParseQuantity(u.instance.Spec.Parameters.Resources.StorageCapacity.String())
+	if err != nil {
+		return false
+	}
+	if *u.persistentVolumeClaim.Spec.Resources.Requests.Storage() != quantity {
+		return true
+	}
+	return false
```

Comment:
resource.Quantity has its own resource comparison method, we should use this.
(not sure if that compiles or you need intermediate variable due to the pointers.

```suggestion
	if u.persistentVolumeClaim.Spec.Resources.Requests.Storage().Equal(*u.instance.Spec.Parameters.Resources.StorageCapacity) {
		return false
	}
	return true
```

Reply:
It needs an intermediate variable. And yes, it's better `Equals`.